src/main/java/com/google/devtools/build/lib/analysis/config/CoreOptions.java

```diff
@@ -342,14 +342,30 @@ public String getTypeDescription() {
       documentationCategory = OptionDocumentationCategory.OUTPUT_PARAMETERS,
       effectTags = {OptionEffectTag.ACTION_COMMAND_LINES},
       help =
-          "Specifies the set of environment variables available to actions. "
+          "Specifies the set of environment variables available to actions with target configuration. "
               + "Variables can be either specified by name, in which case the value will be "
               + "taken from the invocation environment, or by the name=value pair which sets "
               + "the value independent of the invocation environment. This option can be used "
               + "multiple times; for options given for the same variable, the latest wins, options "
               + "for different variables accumulate.")
   public List<Map.Entry<String, String>> actionEnvironment;
 
+  @Option(
+          name = "host_action_env",
+          converter = Converters.OptionalAssignmentConverter.class,
+          allowMultiple = true,
+          defaultValue = "null",
+          documentationCategory = OptionDocumentationCategory.OUTPUT_PARAMETERS,
+          effectTags = {OptionEffectTag.ACTION_COMMAND_LINES},
+          help =
+                  "Specifies the set of environment variables available to actions with host configuration. "
```

Comment:
"with host or execution configurations".

Ideally we'd move away from target/host to non-tool/tool, but we're not there now so no need to burden your PR.